bootstrapgenerator/src/main/java/nl/openweb/hippo/groovy/ScriptClassFactory.java

```diff
@@ -20,67 +20,101 @@
 
+    private ScriptClassFactory() {
+        //No instantiating of this class
+    }
+
+    private static NamespaceMapping getNamespaceResolver() throws NamespaceException {
+        if(!namespaceResolver.hasPrefix("")){
+            namespaceResolver.setMapping("", "");
+        }
+        return namespaceResolver;
+    }
+
+    private static void validateScriptClass(final ScriptClass scriptClass) {
+        if (scriptClass == null || scriptClass.getUpdater() == null) {
+            return;
+        }
+        final String name = scriptClass.getUpdater().name();
+
+        try {
+            NameParser.parse(name, getNamespaceResolver(), nameFactory);
+        } catch (Exception e) {
+            throw new ScriptParseException("Error parsing the updater name for: " + scriptClass.getFile().getAbsolutePath(), e);
+        }
+    }
+
     /**
      * Returns a class that has actually nothing but the Bootstrap and Updater Annotations
      *
      * @param file the file to make a class representation of
      * @return a fake class with the Bootstrap and Updater annotations
      */
     public static ScriptClass getInterpretingClass(final File file) {
-        return  getInterpretingClassStrippingCode(file);
-    }
-
-    public static ScriptClass getInterpretingClassStrippingCode(final File file) {
         groovyClassLoader.clearCache();
         String script;
         try {
-            ensureLinuxLineEnding(file);
-            script = FileUtils.fileRead(file);
+            script = readFileEnsuringLinuxLineEnding(file);
 
             String imports = getAnnotationClasses().stream()
                     .map(clazz -> "import " + clazz.getCanonicalName() + ";")
                     .collect(joining());
 
-            String interpretCode =  imports + script.replaceAll("import .+\n", "")
+            String interpretCode = imports + script.replaceAll("import .+\n", "")
                     .replaceAll("package\\s.*\n", "")
                     .replaceAll("extends\\s.*\\{[^\\u001a]*", "{}");
 
             interpretCode = scrubAnnotations(interpretCode);
-
-            return new ScriptClass(file, groovyClassLoader.parseClass(interpretCode), script);
+            script = stripAnnotations(script);
+            final ScriptClass scriptClass = new ScriptClass(file, groovyClassLoader.parseClass(interpretCode), script);
+            validateScriptClass(scriptClass);
+            return scriptClass;
         } catch (IOException e) {
             return null;
         }
     }
 
-    private static void ensureLinuxLineEnding(final File file) throws IOException {
+    public static String readFileEnsuringLinuxLineEnding(final File file) throws IOException {
         String content = FileUtils.fileRead(file);
-        if(content.contains(LINE_END_WINDOWS)){
-            FileUtils.fileWrite(file, content.replaceAll(LINE_END_WINDOWS, LINE_END_LINUX));
+        if (content.contains(LINE_END_WINDOWS)) {
```

Comment:
Only when there are Windows line endings you try to replace Mac line ending

Reply:
I will fix it